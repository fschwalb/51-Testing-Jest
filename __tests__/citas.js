import Citas from '../js/classes/Citas';


describe( 'Probar la clase de Citas', () => {

    const citas = new Citas();

    const id = Date.now();

    test( 'Agregar una nueva Cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Fernando',
            telefono: '1654321',
            fecha: '10-12-2021',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }

        citas.agregarCita( citaObj );

        // Test
        expect( citas ).toMatchSnapshot();
    });

    test( 'Actualizar Cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Rambo',
            propietario: 'Fernando',
            telefono: '6684414',
            fecha: '15-1-2021',
            hora:'13:40',
            sintomas: 'Solo come'
        };

        citas.editarCita( citaActualizada );

        expect( citas ).toMatchSnapshot();
    });

    test( 'Eliminar Cita', () => {
        citas.eliminarCita( id );

        expect( citas ).toMatchSnapshot();
    });
});