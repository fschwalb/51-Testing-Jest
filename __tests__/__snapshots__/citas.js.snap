// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar Cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "15-1-2021",
      "hora": "13:40",
      "id": 1646084416067,
      "mascota": "Rambo",
      "propietario": "Fernando",
      "sintomas": "Solo come",
      "telefono": "6684414",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva Cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1646084416067,
      "mascota": "Hook",
      "propietario": "Fernando",
      "sintomas": "Solo duerme",
      "telefono": "1654321",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;
